wandb/sdk/internal/sender.py

```diff
@@ -880,6 +880,20 @@ def _sync_spell(self) -> None:
             pass
         # TODO: do something if sync spell is not successful?
 
+    def _setup_fork(self):
+        if self._settings.fork_from:
+            self._resume_state.step = int(self._settings.fork_from.value)
+
+            fork_state = self._api.run_resume_status(
+                entity=self._run.entity,  # type: ignore
+                project_name=self._run.project,
+                name=self._run.run_id,
```

Comment:
Is this correct?

Reply:
yep - its kind of counterintuitive though but it is correct. here is what it is doing: 

1. we call upsert run and create the new run. on the backend, we scan through history and recompute keys info, summary metrics, and history count. 
2. we get this run back and it becomes self._run
3. we need to then get the history count so we can send filestream chunks with the proper offsets. 
4. so we call resume_status *on the same run* to get the newly calculated history count and set it on _resume_state to get the right filestream offset.

Reply:
could you please add documentation here? This is non-trivial and has implicit assumptions about what the backend does, would be good to document

Reply:
Yep it's already there. Added in response to your comment. Sorry should have said so in my comment